drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -0,0 +1,938 @@
+
+	return err;
+}
+
+static int reg_write(struct pmw3360_data *dev_data, u8_t reg, u8_t val)
+{
+	int err;
+
+	__ASSERT_NO_MSG((reg & SPI_WRITE_BIT) == 0);
+
+	dev_data->last_read_burst = false;
+
+	err = spi_cs_ctrl(dev_data, true);
+	if (err) {
+		goto error;
+	}
+
+	u8_t buf[] = {
+		SPI_WRITE_BIT | reg,
+		val
+	};
+	const struct spi_buf tx_buf = {
+		.buf = buf,
+		.len = ARRAY_SIZE(buf)
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+
+	err = spi_transceive(dev_data->spi_dev, &spi_cfg, &tx, NULL);
+	if (err) {
+		goto error;
+	}
+
+	k_busy_wait(T_SCLK_NCS_WR);
+
+	err = spi_cs_ctrl(dev_data, false);
+	if (err) {
+		goto error;
+	}
+
+	k_busy_wait(T_SWX);
+
+	return 0;
+
+error:
+	LOG_ERR("SPI reg write failed");
+
+	return err;
+}
+
+static int motion_burst_read(struct pmw3360_data *dev_data, u8_t *data,
+			     size_t burst_size)
+{
+	int err;
+
+	__ASSERT_NO_MSG(burst_size <= PMW3360_MAX_BURST_SIZE);
+
+	/* Write any value to motion burst register only if there have been
+	 * other SPI transmissions with sensor since last burst read.
+	 */
+	if (!dev_data->last_read_burst) {
+		err = reg_write(dev_data, PMW3360_REG_MOTION_BURST, 0x00);
+		if (err) {
+			goto error;
+		}
+		dev_data->last_read_burst = true;
+	}
+
+	err = spi_cs_ctrl(dev_data, true);
+	if (err) {
+		goto error;
+	}
+
+	/* Send motion burst address */
+	u8_t reg_buf[] = {
+		PMW3360_REG_MOTION_BURST
+	};
+	const struct spi_buf tx_buf = {
+		.buf = reg_buf,
+		.len = ARRAY_SIZE(reg_buf)
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+
+	err = spi_transceive(dev_data->spi_dev, &spi_cfg, &tx, NULL);
+	if (err) {
+		goto error;
+	}
+
+	k_busy_wait(T_SRAD_MOTBR);
+
+	/* Read up to 12 bytes in burst */
+	const struct spi_buf rx_buf = {
+		.buf = data,
+		.len = burst_size,
+	};
+	const struct spi_buf_set rx = {
+		.buffers = &rx_buf,
+		.count = 1
+	};
+
+	err = spi_transceive(dev_data->spi_dev, &spi_cfg, NULL, &rx);
+	if (err) {
+		goto error;
+	}
+
+	/* Terminate burst */
+	err = spi_cs_ctrl(dev_data, false);
+	if (err) {
+		goto error;
+	}
+	k_busy_wait(T_BEXIT);
+
+	return 0;
+
+error:
+	LOG_ERR("SPI burst write failed");
+
+	return err;
+}
+
+static int burst_write(struct pmw3360_data *dev_data, u8_t reg, const u8_t *buf,
+		       size_t size)
+{
+	int err;
+
+	dev_data->last_read_burst = false;
+	err = spi_cs_ctrl(dev_data, true);
+	if (err) {
+		goto error;
+	}
+
+	/* Write address of burst register */
+	u8_t write_buf = reg | SPI_WRITE_BIT;
+	struct spi_buf tx_buf = {
+		.buf = &write_buf,
+		.len = 1
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+
+	err = spi_transceive(dev_data->spi_dev, &spi_cfg, &tx, NULL);
+	if (err) {
+		goto error;
+	}
+
+	/* Write data */
+	for (size_t i = 0; i < size; i++) {
+		write_buf = buf[i];
+
+		err = spi_transceive(dev_data->spi_dev, &spi_cfg, &tx, NULL);
+		if (err) {
+			goto error;
+		}
+
+		k_busy_wait(15);
```

Comment:
I guess this was because this timing is not defined in the table and does not have any name. It only appears in the figure. We should avoid magic numbers - I have given it a name and added, along with a comment, to other timings.